deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -246,4 +257,14 @@ Duration asDuration() {
             return valueNanos <= 0 ? Duration.ZERO : Duration.ofNanos(valueNanos);
         }
     }
+
+    interface DeadlineClock {
+        default long nanoTime() {
+            return System.nanoTime();
+        }
```

Comment:
imo no sense in making this a default method since it's the entire point of the deadline clock. We could even drop `DefaultClock` in favor of updating the initializer to `private static DeadlineClock clock = System::nanoTime;`